### PR TITLE
Mbst 13131 configure cassandra connection timeouts

### PR DESCRIPTION
**Pull request by cognitive-i**
Extracted builder from DataStax to permit configuration of timeouts via environment.properties

### CODE REVIEW

File: atlas-api/src/main/java/org/atlasapi/AtlasPersistenceModule.java [hunk 1]
```diff
@@ -6,6 +6,10 @@
 
 import javax.annotation.PostConstruct;
 
+import com.datastax.driver.core.*;
```
Comment (emilssolmanis) [inline atlas-api/src/main/java/org/atlasapi/AtlasPersistenceModule.java]: Misonfigged IntelliJ? We don't normally do star imports.

File: atlas-api/src/main/java/org/atlasapi/AtlasPersistenceModule.java [hunk 2]
```diff
@@ -120,6 +126,35 @@ public void init() {
...
     }
 
+
+    private DatastaxCassandraService getCassandraService(Iterable<String> nodes,
+                                                         Integer connectionsPerHostLocal, Integer connectionsPerHostRemote,
+                                                         Integer connectionTimeoutMs, Integer readTimeoutMs) {
+        // this code is borrowed from DatastaxCassandraService::DatastaxCassandraService
+        // but enhances timeout criteria to improve Atlas performance when cluster is loaded
+        PoolingOptions poolingOptions = new PoolingOptions();
+        poolingOptions.setMaxConnectionsPerHost(HostDistance.LOCAL, Preconditions.checkNotNull(connectionsPerHostLocal));
```
Comment (emilssolmanis) [inline atlas-api/src/main/java/org/atlasapi/AtlasPersistenceModule.java]: The methods on `Preconditions` are usually statically imported.

File: atlas-api/src/main/java/org/atlasapi/AtlasPersistenceModule.java [hunk 3]
```diff
@@ -120,6 +126,35 @@ public void init() {
...
+        poolingOptions.setCoreConnectionsPerHost(HostDistance.LOCAL, Preconditions.checkNotNull(connectionsPerHostLocal));
+        poolingOptions.setMaxConnectionsPerHost(HostDistance.REMOTE, Preconditions.checkNotNull(connectionsPerHostRemote));
+        poolingOptions.setCoreConnectionsPerHost(HostDistance.REMOTE, Preconditions.checkNotNull(connectionsPerHostRemote));
+
+        Cluster.Builder builder = Cluster.builder()
+                .addContactPoints((String[]) FluentIterable.from(nodes).toArray(String.class))
+                .withCompression(ProtocolOptions.Compression.SNAPPY)
```
Comment (emilssolmanis) [inline atlas-api/src/main/java/org/atlasapi/AtlasPersistenceModule.java]: Is compression a client-only setting in C* or does it require server config too? If the latter, have we done that config?

Comment (cognitive-i) [inline atlas-api/src/main/java/org/atlasapi/AtlasPersistenceModule.java]: This replication of the existing DatastaxCService.  @Dalamar42 has pointed out that we control DatastaxCassandraService so I'm going to refactor these changes to that project and throw away the replication

**Pull request closed without merging**